File: python/revoke.py
"""Revoke builder fee approval."""

from client import exchange, sign_hash

res = exchange({
    "action": {"type": "approveBuilderFee", "maxFeeRate": "0%"},
})
sig = sign_hash(res["hash"])

exchange({
    "action": {"type": "approveBuilderFee", "maxFeeRate": "0%"},
    "nonce": res["nonce"],
    "signature": sig,
})

print("Builder fee revoked.")

File: typescript/src/revoke.ts
import { exchange, signHash } from "./client";

async function main() {
  const res = await exchange({
    action: { type: "approveBuilderFee", maxFeeRate: "0%" },
  });
  const sig = await signHash(res.hash);

  await exchange({
    action: { type: "approveBuilderFee", maxFeeRate: "0%" },
    nonce: res.nonce,
    signature: sig,
  });

  console.log("Builder fee revoked.");
}

main();
